packages/utils/src/dal/mikro-orm/integration-tests/__tests__/mikro-orm-repository.spec.ts

```diff
@@ -0,0 +1,844 @@
+        relationsToUpsert: ["entity1"],
+        relationsToSkip: [],
+      })
+
+      entity2.title = "newen2"
+      entity2.entity1.title = "newen1"
+      await manager2().upsertWithReplace([entity2], {
+        relationsToUpsert: [],
+        relationsToSkip: ["entity1"],
+      })
+      const listedEntities = await manager2().find({
+        where: { id: "2" },
+        options: { populate: ["entity1"] },
+      })
+
+      expect(listedEntities).toHaveLength(1)
+      expect(listedEntities[0]).toEqual(
+        expect.objectContaining({
+          id: "2",
+          title: "newen2",
+          entity1: expect.objectContaining({
+            title: "en1",
+          }),
+        })
+      )
+    })
+
+    it("should successfully update an entity with a sub-entity many-to-one relation", async () => {
+      const entity2 = {
+        id: "2",
+        title: "en2",
+        entity1: { title: "en1" },
+      }
+
+      await manager2().upsertWithReplace([entity2], {
+        relationsToUpsert: ["entity1"],
+        relationsToSkip: [],
+      })
+
+      const [createEntity2] = await manager2().find({
+        where: { id: "2" },
+        options: { populate: ["entity1"] },
+      })
+
+      createEntity2.title = "newen2"
+      createEntity2.entity1!.title = "newen1"
+
+      await manager2().upsertWithReplace([createEntity2], {
+        relationsToUpsert: ["entity1"],
+        relationsToSkip: [],
+      })
+
+      const [updatedEntity2] = await manager2().find({
+        where: { id: "2" },
+        options: { populate: ["entity1"] },
+      })
+
+      expect(wrap(updatedEntity2).toPOJO()).toEqual(
+        expect.objectContaining({
+          id: "2",
+          title: "newen2",
+          entity1_id: expect.any(String),
+          entity1: expect.objectContaining({
+            id: expect.any(String),
+            title: "newen1",
+          }),
+        })
+      )
+    })
+
+    it("should successfully unset an entity of a many-to-one relation", async () => {
+      const entity2 = {
+        id: "2",
+        title: "en2",
+        entity1: { title: "en1" } as any,
+      }
+
+      await manager2().upsertWithReplace([entity2], {
+        relationsToUpsert: ["entity1"],
+        relationsToSkip: [],
+      })
+      entity2.title = "newen2"
+      entity2.entity1 = null
+      await manager2().upsertWithReplace([entity2], {
+        relationsToUpsert: ["entity1"],
+        relationsToSkip: [],
+      })
+      const listedEntities = await manager2().find({
+        where: { id: "2" },
+        options: { populate: ["entity1"] },
+      })
+
+      expect(listedEntities).toHaveLength(1)
+      expect(listedEntities[0]).toEqual(
+        expect.objectContaining({
+          id: "2",
+          title: "newen2",
+          entity1: null,
+        })
+      )
+    })
+
+    it("should only create the parent entity of a one-to-many if relation is not included", async () => {
+      const entity1 = {
+        id: "1",
```

Comment:
q: if an ID is passed and is not present in the db, we seem to be creating it. Should we guard against this case and throw in the event of an incorrect ID being passed?

Reply:
it should create it, we are by design allowing to have an id provided when we go through the generateEntityId, I would say that it should be allowed if for example the user does not want to use the same id generation pattern wdyt?

Reply:
@riqwan In the workflows there are scenarios where you want to recreate a deleted entry, which is why we allowed passing an ID in the modules (but it shouldn't be exposed in the HTTP endpoints). I think this is a bigger discussion that I'll leave for another time